databricks_cli/secrets/cli.py

```
@@ -0,0 +1,310 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import base64
+import click
+from tabulate import tabulate
+
+from databricks_cli.click_types import OutputClickType, SecretScopeClickType, SecretKeyClickType, \
+    SecretPrincipalClickType
+from databricks_cli.secrets.api import SecretApi
+from databricks_cli.utils import eat_exceptions, CONTEXT_SETTINGS, pretty_format, truncate_string, \
+    error_and_quit
+from databricks_cli.configure.config import provide_api_client, profile_option
+from databricks_cli.version import print_version_callback, version
+
+
+SCOPE_HEADER = ('Scope', 'Backend')
+SECRET_HEADER = ('Key name', 'Last updated')
+ACL_HEADER = ('Principal', 'Permission')
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help="Creates a secret scope.")
+@click.option('--scope', required=True, type=SecretScopeClickType(), help=SecretScopeClickType.help)
+@click.option('--initial-manage-principal',
+              help='The initial principal that can manage the created secret scope.'
+              ' If specified, the initial ACL with MANAGE permission applied to the scope is'
+              ' assigned to the supplied principal (user or group). Currently, the only supported'
+              ' principal for this option is the group "users", which contains all users in the'
+              ' workspace. If not specified, the initial ACL with MANAGE permission applied to the'
+              ' scope is assigned to the request issuer\'s user identity.')
+@profile_option
+@eat_exceptions
+@provide_api_client
+def create_scope(api_client, scope, initial_manage_principal):
+    """
+    Creates a new secret scope with given name.
+    """
+    SecretApi(api_client).create_scope(scope, initial_manage_principal)
```

Why does
```
databricks secrets create-scope --scope andrew --initial-manage-principal apple
```
not get rejected by the backend?

---

I think the backend change is not fully rolled out.

---

Proto silently ignores invalid JSON fields too :).